libr/anal/p/anal_x86_cs.c

```diff
@@ -2172,7 +2172,6 @@ static void anop(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len, csh
 				op->stackptr = regsz;
 				op->type |= R_ANAL_OP_TYPE_REG;
 			}
-			op->ptr = INSOP(1).imm;
```

Comment:
Yup this looks wrong , but the problem is there are lot of places where op->ptr and op->val is used interchangebly , let me try a proper fix for this !